source/simulator/core.cc

```diff
@@ -366,6 +370,13 @@ namespace aspect
     mesh_refinement_manager.initialize_simulator (*this);
     mesh_refinement_manager.parse_parameters (prm);
 
+    // VoF Must be initialized after mesh_refinement_manager, due to needing to check
+    // for a mesh refinement strategy
+    if (parameters.volume_of_fluid_tracking_enabled)
+      {
+        volume_of_fluid_handler->initialize (prm);
```

Comment:
I was a bit confused by you calling `initialize` here without ever calling `initialize_simulator` and `parse_parameters`, but that was before I saw that you do this in the constructor of the VoFHandler. Would there be a problem moving the calls to `initialize_simulator` and `parse_parameters` here, and move the initialization of the `assembler` member of VoFHandler into the `initialize` function instead of the constructor? That would be more consistent with the other plugin systems.

Reply:
One issue is that for the current construction, the parameters must be parsed before the introspection object is created, which is done at construction of the Simulator object.